ftw/book/layer.py

```diff
@@ -11,11 +12,29 @@ class BookTraverse(DefaultPublishTraverse):
 
     def publishTraverse(self, request, name):
 
-        layout_layer_name = getattr(self.context, 'latex_layout', '')
-        if layout_layer_name:
-            layout_layer = resolve(layout_layer_name)
-            provide_request_layer(request, layout_layer)
-
-        provide_request_layer(request, IWithinBookLayer)
+        provideBookLayers(self.context, request)
 
         return super(BookTraverse, self).publishTraverse(request, name)
+
+
+def provideBookLayers(context, request):
+    layout_layer_name = getattr(context, 'latex_layout', '')
+    if layout_layer_name:
+        layout_layer = resolve(layout_layer_name)
+        provide_request_layer(request, layout_layer)
+
+    provide_request_layer(request, IWithinBookLayer)
+
+
+class BookContext(object):
```

Comment:
I'd also rename this class.
I'd probably also change it to a function using the ``contextmanager`` decorator (``contextlib``), e.g.:

```python
@contextmanager
def providing_book_layers(book, request):
    original_interfaces = ..
    try:
        provide_book_layers...
    finally:
        .. reset
```

I think this would be more expressive and nicer when it is used:
```python
with providing_book_layers(book, request):
    # do stuff
```